src/PromiseCore.php

```diff
@@ -80,7 +80,7 @@ public function __construct(
         $handle,
         ResponseBuilder $responseBuilder
     ) {
-        if (!is_resource($handle)) {
+        if (!is_resource($handle) && !(is_object($handle) && $handle instanceof \GdImage)) {
```

Comment:
i am confused by the second part, i assume you meant CurlHandle rather than GdImage?

but i think we should make an explicit check for the php version. something like 

```
if (
    PHP_MAJOR_VERSION === 7 && !is_resource($handle)
    || PHP_MAJOR_VERSION > 7 && !$handle instanceof CurlHandle
) {
...
```